### PR TITLE
support client_credentials grant_type

### PR DESCRIPTION
**Pull request by jbolda**
## Motivation

Add specific support for additional `grant_type`s. This was required to enable m2m token use. Note, this flow does not appear to run the rules, so the return logic is slightly adjusted to suit.

### CODE REVIEW

**Review by dagda1 — CHANGES_REQUESTED** [review]

File: packages/auth0/src/handlers/auth0-handlers.ts [hunk 1]
```diff
@@ -153,19 +153,35 @@ export const createAuth0Handlers = (store: Auth0Store, people: Iterable<Person>,
...
       let username: string;
       let password: string | undefined;
-      let response_client_id: string;
-      let response_audience: string;
+      let responseClientId: string = req?.body?.client_id as string ?? clientID;
+      let responseAudience: string = req?.body?.audience as string ?? audience;
 
+      // grant_type list as defined by auth0
+      // https://auth0.com/docs/get-started/applications/application-grant-types#spec-conforming-grants
       if (grant_type === 'password') {
```
[the remark this answers is not in the document]
Comment (jbolda) [reply packages/auth0/src/handlers/auth0-handlers.ts:161]: Yea, that makes a lot of sense. 👍

File: packages/auth0/src/handlers/auth0-handlers.ts [hunk 2]
```diff
@@ -153,19 +153,35 @@ export const createAuth0Handlers = (store: Auth0Store, people: Iterable<Person>,
...
         username = req.body.username;
         password = req.body.password;
-        response_client_id = req?.body?.client_id as string || clientID;
-        response_audience = req?.body?.audience as string|| audience;
+      } else if (grant_type === 'client_credentials') {
```
[the remark this answers is not in the document]
Comment (jbolda) [reply packages/auth0/src/handlers/auth0-handlers.ts:164]: Yea I can see that. I suspect that we will end up fleshing out these `grant_type`s more to cover them all. I am tempted to say that maybe it makes sense to do it at that time?

[the remark this answers is not in the document]
Comment (dagda1) [reply packages/auth0/src/handlers/auth0-handlers.ts:164]: it's up to you but I would do it now and you'll save time in the long run or else it'll be really hard to tell what is going on.

I could do it in another PR if you want when you are finished

[the remark this answers is not in the document]
Comment (jbolda) [reply packages/auth0/src/handlers/auth0-handlers.ts:164]: Might as well do it as part of this PR then. I can work on that on Monday.

[the remark this answers is not in the document]
Comment (jbolda) [reply packages/auth0/src/handlers/auth0-handlers.ts]: I feel pretty good about the refactor: https://github.com/thefrontside/simulacrum/pull/230/commits/134e5fe1ef2bd0fae316b0b23b7a2181e0148e73

**Review by dagda1 — APPROVED** [review]

**Pull request merged**